salt/states/win_mslicense.py

```diff
@@ -0,0 +1,541 @@
+"""
+Module for managing OS and software licenses on Windows,
+utilizing PowerShell and CIM methods.
+
+Invoke-CimMethod is supported in PowerShell 3.0, so
+the module can potentially work starting from Windows 7.
+
+Guaranteed tests were conducted on Windows 10.
+"""
+import logging
+import salt.utils.platform
+
+log = logging.getLogger(__name__)
+__virtualname__ = "mslicense"
+
+
+def __virtual__():
+    """
+    Only work on Windows
+    """
+    if salt.utils.platform.is_windows():
+        return __virtualname__
+    return (False, "Only Windows OS supported")
+
+
+def present(name: str, kms_host="", kms_port=0, activate=False):
+    """
+    Ensure that the product key is present and optionally activate it.
+
+    name
+        Product key for OS or software
+    kms_host
+        Host for KMS activation
+    kms_port
+        Port for KMS activation
+    activate
+        Whether to activate the product
+    """
+
+    name = name.upper()
+    kms_host = kms_host.lower()
+
+    ret = {"name": name[-5:], "result": None, "comment": "", "changes": {}}
+
+    def infoIsNone(t: tuple, k: str): return None if (
```

Comment:
We don't use camelCase in Salt unless we're working with classes. You should probably make this a hidden function as well by preceding it with an underscore (`_`) so it is not exposed in the documentation. Please use `_info_is_none`.